internal/pkg/util/fs/helper_linux_test.go

```diff
@@ -16,6 +16,125 @@ import (
 	"github.com/sylabs/singularity/internal/pkg/test"
 )
 
+func TestEnsureFileWithPermission(t *testing.T) {
+	test.DropPrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	tmpDir, err := ioutil.TempDir("", "ensure_file_perm-")
+	if err != nil {
+		t.Errorf("Unable to make tmpdir %s", err)
+	}
+
+	//
+	// First test: Ensure a already-existing file is the
+	// correct permssion.
+	//
+
+	existFile := filepath.Join(tmpDir, "already-exists")
+
+	// Create the test file.
+	fp, err := os.OpenFile(existFile, os.O_CREATE, 0755)
```

Comment:
This test will fail on a system with a non-standard umask, as 0755 is before umask. Adding a separate `chmod` should make it safer.